### PR TITLE
LPS-196387

### PR DESCRIPTION
**Pull request by ces-totrinh**
Hi @ces-quanhuynh @ces-luannguyen @locpham97 ,
Please help me review this ticket:
https://liferay.atlassian.net/browse/LPS-196387.
Notes:
Document tabs will be displayed after the first document is uploaded and child documents' resource permissions are initialized (see https://liferay.atlassian.net/browse/LPS-196388)
Thanks,

### CODE REVIEW

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/configuration/permission/propagation/DLFolderPortletConfigurationPermissionPropagation.java [hunk 1]
```diff
@@ -40,18 +45,36 @@ public String getMessage() {
 			"folder";
 	}
 
+	@Override
+	public boolean isAvailable(PortletRequest portletRequest) {
+		long[] classPKs = getClassPKs(portletRequest);
+
+		if ((classPKs.length == 1) & super.isAvailable(portletRequest)) {
```
Comment (ces-luannguyen) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/configuration/permission/propagation/DLFolderPortletConfigurationPermissionPropagation.java:90]: &&

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/configuration/permission/propagation/DLFolderPortletConfigurationPermissionPropagation.java [hunk 2]
```diff
@@ -62,8 +123,26 @@ protected boolean getDefaultPermissionPropagation(
...
 		};
 	}
 
+	private long _getClassPK(PortletRequest portletRequest) {
+		long[] resourcePrimKeys = getClassPKs(portletRequest);
+
+		return resourcePrimKeys[0];
+	}
+
+	private static final String _CHILD_DOCUMENTS_RESOURCE_NAME =
```
Comment (locpham97) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/configuration/permission/propagation/DLFolderPortletConfigurationPermissionPropagation.java:137]: _CHILD_DOCUMENT_RESOURCE_NAME or simple  _CHILD_RESOURCE_NAME

**Comment by ces-locpham** [comment]
It is LGTM. Just one thing about a variable that I mentioned in the previous comment.

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/configuration/permission/propagation/DLFolderPortletConfigurationPermissionPropagation.java (hunk 2 above)
Comment (tototrinh) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/configuration/permission/propagation/DLFolderPortletConfigurationPermissionPropagation.java:137]: I think _CHILD_DOCUMENT_RESOURCE_NAME would be better.

Comment (tototrinh) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/configuration/permission/propagation/DLFolderPortletConfigurationPermissionPropagation.java:137]: Sorry, this tab is defined for the child documents uploaded to the folder so I think it should be `_CHILD_DOCUMENTS_RESOURCE_NAME`. I will change the tab label to `documents permissions`. Thanks.

**Comment by ces-totrinh** [comment]
I will close this PR. See: https://github.com/tototrinh/liferay-portal/pull/79

**Pull request closed without merging**